language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/inlayhint/InlayHintProvider.java

```diff
@@ -84,7 +100,7 @@ public static List<InlayHint> getInlayHint(InlayHintContext context) {
             for (int argumentIndex = 0; argumentIndex < argList.size(); argumentIndex++) {
                 //Find the inlay-hint location for the argument
                 NonTerminalNode argument = argList.get(argumentIndex);
-                if (argument.kind() != SyntaxKind.POSITIONAL_ARG) {
+                if (argument.kind() == SyntaxKind.NAMED_ARG) {
```

Comment:
Assume that the currentArg statisfies `argument.kind() == SyntaxKind.REST_ARG` condition.  we should not show any inlayHints In the following case.

```ballerina
type Person record {|
    string name;
    int age;
|};

public function main() {
    Person p = {
        name: "",
        age: 0
    };
    string testResult = test(10, ...p);
}

function test(int kind, string name, int age) returns string {
    return "Hello";
}
```

However, we can suggests inlay hints in the following case.

```ballerina

public function main() {
    int[] arr = [1, 2, 3];
    string testResult = test("ballerina", ...arr );
}

function test(string kind, int... values) returns string {
    return "Hello";
}
```

Reply:
Fixed with [b5bce24](https://github.com/ballerina-platform/ballerina-lang/pull/40868/commits/b5bce24516ecb7058f23f72f193fad09d92fa047)